docs/articles/advanced_topics/generic_host.md

```diff
@@ -0,0 +1,141 @@
+
+It allows developers to build console applications, background services, and other types of .NET applications that can run as standalone processes, Windows services, or Docker containers, among other deployment options. By using a generic host, developers can focus on implementing the core business logic of their application rather than dealing with the infrastructure and plumbing required to host and manage it.
+
+You can read more about Generic hosts [here](https://learn.microsoft.com/en-us/dotnet/core/extensions/generic-host).
+
+# Making a Bot using the Generic Host
+## Setting up the Builder
+To get started, you'll need to write some code that configures and runs your application. Here's a simple example that shows how to use the Generic Host to run a bot service:
+```cs
+private static async Task Main()
+{
+    await Host.CreateDefaultBuilder()
+      .UseConsoleLifetime()
+      .ConfigureServices((hostContext, services) =>
+      {
+          services.AddHostedService<YourBotHost>();
+      })
+      .RunConsoleAsync();
+}
+```
+This code does a few things. First, it calls the `Host.CreateDefaultBuilder()` method, which creates a new `IHostBuilder` instance with some default settings. Then, it calls the `UseConsoleLifetime()` method to configure the lifetime of the host. This tells the host to keep running until it receives a [SIGINT Signal](https://en.wikipedia.org/wiki/Signal_(IPC)#SIGINT) i.e. when the user presses Ctrl+C in the console, or when another program tells it to stop.
+
+Next, it configures the services that the host will use by calling the `ConfigureServices()` method. In this case, it adds a new `YourBotHost` service, which is a class that you'll need to define next.
+
+Finally, it calls the `RunConsoleAsync()` method to start the host and begin running your service. That's it! With just a few lines of code, you can use the .NET Generic Host to run your application as a service.
+
+## Setting up your Service
+You'll need to create a class which implements the `IHostedService` interface, which defines two methods: `StartAsync()` and `StopAsync()`. These methods are called by the host when your application starts and stops, respectively.
+
+```cs
+public sealed class YourBotHost : IHostedService
+{
+    private readonly ILogger<YourBotHost> _logger;
+    private readonly IHostApplicationLifetime _applicationLifetime;
+    private readonly DiscordClient _discordClient;
+
+    public YourBotHost(ILogger<YourBotHost> logger, IHostApplicationLifetime applicationLifetime)
+    {
+        this._logger = logger;
+        this._applicationLifetime = applicationLifetime;
+        this._discordClient = new(new()
+        {
+            Token = "YourBotTokenHere",
+            TokenType = TokenType.Bot,
+            Intents = DiscordIntents.AllUnprivileged
+        });
+    }
```

Comment:
I'd suggest adding a note in regards to hard-coded tokens; it's mentioned in some prior article, but people are unfortunately prone to copy-pasting things.

Something along the lines of:

> **Warning**
> Hard-coding tokens is a **bad practice** and only used for exemplary purposes. In production code, using DI to inject a configuration, and pulling from that instead. [\[Docs\]](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-7.0)

Reply:
I think what's there is fine; just the supplementary note that needs to be added

Reply:
As per the docs I linked, the prefered way to get config variables is via `IConfiguration` and DI